### PR TITLE
Re-introduce the async API

### LINKED ISSUE #148: `async`/`await` support
Initially this client was asynchronous. We rewrote it to become synchronous when introducing viaduct.

We could offer two implementations to satisfy both paradigms

### PR DESCRIPTION
**Pull request by Dexterp37**
Fixes #148 

This does not retain the sync API.

### CODE REVIEW

File: src/client/signatures.rs [hunk 1]
```diff
@@ -70,8 +71,9 @@ fn epoch_seconds() -> u64 {
 ///    .build();
 /// # }
 /// ```
-pub trait Verification: Send {
-    fn fetch_certificate_chain(
+#[async_trait]
+pub trait Verification: Send + Sync {
+    async fn fetch_certificate_chain(
```
Comment (Dexterp37) [inline src/client/signatures.rs:76]: @leplatrem should all the methods of this trait be `async`?

**Review by leplatrem — APPROVED** [review]
This looks good to me 

By dropping the sync support, we lose the potential integration into [application-services](https://github.com/mozilla/application-services). I'm OK with that. Our initial use-cases of @ncalexan's windows agent and @dmose's Nimbus took a different direction, and shouldn't prevent us to move on here.

We can revisit the idea of having both sync and async APIs using `tokio::runtime::Handle` spawn or similar, when concrete use-cases will show up.

File: src/client/signatures.rs (hunk 1 above)
Comment (Dexterp37) [reply src/client/signatures.rs:76]: > I believe `fetch_certificate_chain()` and thus `verify()` is enough.
> 
> I don't think serializing data and calling crypto primitives should be. If some day we find ourselves in a situation where they should, then we'll change it. What do you think?

@leplatrem I agree. We can cross that bridge once we see a problem with the current approach :-)

I'll fix the test and merge this :)

**Review by leplatrem — APPROVED** [review]
:tada:

**Comment by Dexterp37** [comment]
> 🎉

Whoops, I forgot to update the README, but it's done now :-P

**Pull request merged**